src/Datasource/QueryInterface.php

```diff
@@ -40,7 +45,10 @@ interface QueryInterface
      * @param bool $overwrite whether to reset fields with passed list or not
      * @return $this
      */
-    public function select($fields, bool $overwrite = false);
+    public function select(
+        ExpressionInterface|Association|Table|callable|array|string $fields,
```

Comment:
"I heard ya like types, here ya go, have half a dozen."

Reply:
`Datasource` package is using classes from `Database` and `ORM` package :(

Reply:
This datasource package is looking less useful.

Reply:
Changed QueryInterface to type hint only non-database types and then added them in the subclasses.

Reply:
Added `int` because we test selecting an integer.